core/src/main/kotlin/org/evomaster/core/output/dto/DtoWriter.kt

```diff
@@ -102,8 +104,16 @@ object DtoWriter {
             is DateTimeGene -> "String"
             is RegexGene -> "String"
             is BooleanGene -> "Boolean"
-            is ObjectGene -> StringUtils.capitalization(fieldName)
+            is ObjectGene -> field.refType?:StringUtils.capitalization(fieldName)
             else -> throw Exception("Not supported gene at the moment: ${field?.javaClass?.simpleName}")
         })
     }
+
+    /**
+     * @return collected DTOs by the DTO writer. Only for testing, otherwise we're breaking encapsulation.
+     */
+    @VisibleForTesting
+    internal fun getCollectedDtos() : Map<String, DtoClass> {
+        return dtoCollector
```

Comment:
i didn't realize it before, but isn't `dtoCollector` a global mutable static state? this create possible issues. either should be removed (and passed as input between the function that first instantiate and then use it), or this class should be refactored into a singleton service

Reply:
No, but it used to be. `DtoWriter` is now a class instead of an object, so the `dtoCollector` is now created every time `DtoWriter()` is instantiated. I actually faced that issue when I started writing my code, realized and changed from object to class

Reply:
ah, make sense